model.py

```diff
@@ -0,0 +1,1232 @@
+            elif self.table_condition is not None and not cell_value.startswith(self.table_condition):
+                entry.configure(bg='red')
+            elif self.table_condition in cell_value:
+                entry.configure(bg='green')
+            else:
+                entry.configure(bg='red')
+
+            seen_values.add(cell_value)
+            self.original_table_values[i] = cell_value
+
+        return all_entries_valid
+
+    def validate_numer_pojemnika(self, event=None):
+        numer_pojemnika_value = self.entry_container.get()
+        if not numer_pojemnika_value or len(numer_pojemnika_value) != 9:
+            messagebox.showwarning("Ostrzeżenie", "Numer pojemnika powinien zawierać dokładnie 9 znaków!")
+            self.entry_container.delete(0, tk.END)
+            return False
+        else:
+            return True
+
+    def weryfikuj_dane(self, prefix, entry, rubryka):
+        if not entry.get().startswith(prefix):
+            messagebox.showwarning("Ostrzeżenie", f"Niepoprawne dane w rubryce {rubryka}!")
+            entry.delete(0, tk.END)
+
+    def dodaj_dane(self):
+        for entry in self.table_entries:
+            if entry.cget("background") == 'yellow':
+                entry.configure(bg='yellow')  # Wartość powtórzona
+                self.button_add["state"] = "disabled"
+                return
+
+        self.weryfikuj_dane("R", self.entry_container, "Numer pojemnika")
+        self.weryfikuj_dane("30S", self.entry_pn, "PN")
+
+        pn = self.entry_pn.get()
+        numer_pojemnika = self.entry_container.get()
+        ilosc_sztuk = self.entry_quantity.get()
+        powod_nok = self.entry_powod_nok.get()
+        dzialania = self.slider_dzialania.get()
+
+        data = time.strftime('%Y-%m-%d')
+        godzina = time.strftime('%H:%M:%S')
+
+        if not pn or not numer_pojemnika or not ilosc_sztuk or not powod_nok or not dzialania:
+            messagebox.showwarning("Ostrzeżenie", "Wypełnij wszystkie pola!")
+            return
+
+        self.dane = {"PN": [], "Numer pojemnika": [], "Ilość sztuk": [], "Powód sztuki NOK": [],
+                     "Data": [], "Godzina": [], "Działania": [], "Tabela": []}
+        self.dane["PN"].append(pn)
+        self.dane["Numer pojemnika"].append(numer_pojemnika)
+        self.dane["Ilość sztuk"].append(ilosc_sztuk)
+        self.dane["Powód sztuki NOK"].append(powod_nok)
+        self.dane["Działania"].append(dzialania)
+        self.dane["Data"].append(data)
+        self.dane["Godzina"].append(godzina)
+        tabela = [entry.get() for entry in self.table_entries]
+        self.dane["Tabela"].append(tabela)
+
+        self.generuj_raport()
+
+        self.button_add["state"] = "disabled"
+        self.button_verify["state"] = "normal"
+
+        self.entry_pn.delete(0, tk.END)
+        self.entry_container.delete(0, tk.END)
+        self.entry_quantity.delete(0, tk.END)
+        self.entry_powod_nok.delete(0, tk.END)

```

Comment:
dla tych obiektów entry które możesz stworzyć zrób własną metodę delete i niech nie Beirze nic jako parametr (chyba ze jest jakoś przypadek gdzie potrzebujesz czegoś). i w tej Twojej metodzie daj, żeby robiło to co masz tutaj. Nie podoba mi się to 0, tk.END i to że musisz to zabić 4 razy, zastanów się co można z tym zrobić